python/user.py

```diff
@@ -0,0 +1,36 @@
+import os
+import sys
+import glob
+
+def startup(userpythonhome):
+    """
+    Run startup logic for this QGIS user
+    """
+    try:
+        # Start up can be a startup.py or a startup package
+        import startup
+    except ImportError:
+        pass 
+
+    expressionspath = os.path.join(userpythonhome, "expressions")
+    load_user_expressions([expressionspath])
+
+
+def load_user_expressions(paths):
```

Comment:
Not sure about this one...

I would prefer not to add more paths to sys.path (I think it is a bad practice that pollutes the global namespace). Could we just have some generic approach where the user.py would make all subdirs (python packages) inside ~/.qgis2/python appear like they were located under qgis.user module? E.g. ~/.qgis2/python/expressions would be available as qgis.user.expressions

Then specific things like loading expression functions could be as simple as "from qgis.user.expressions import *" ?

Instead of "import startup" it seems that execfile() may be a better choice...